### PR TITLE
top: add support for quitting top by pressing q

### PR DESCRIPTION
**Pull request by jamesmintram**
This PR adds support for quitting `top` by pressing q. This was the behaviour I expected based on other "top"s

### CODE REVIEW

File: Userland/Utilities/top.cpp [hunk 1]
```diff
@@ -177,6 +177,28 @@ static void parse_args(Main::Arguments arguments, TopOption& top_option)
     args_parser.parse(arguments);
 }
 
+static bool check_quit()
+{
+    char c = '\0';
+    read(STDIN_FILENO, &c, sizeof(c));
+    return c == 'q' || c == 'Q';
+    return true;
```
Comment (linusg) [inline Userland/Utilities/top.cpp:185]: ?

File: Userland/Utilities/top.cpp [hunk 2]
```diff
@@ -277,6 +301,12 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
         }
         threads.clear_with_capacity();
         prev = move(current);
-        sleep(top_option.delay_time);
+
+        for (int sleep_slice = 0; sleep_slice < top_option.delay_time * 1000; sleep_slice += 100) {
+            if (check_quit()) {
+                exit(0);
+            }
```
Comment (linusg) [inline Userland/Utilities/top.cpp:308]: Please omit curly braces for single-line bodies

**Review by linusg — CHANGES_REQUESTED** [review]

File: Userland/Utilities/top.cpp (hunk 1 above)
Comment (jamesmintram) [reply Userland/Utilities/top.cpp:185]: Good catch :-O

**Review by linusg — COMMENTED** [review]
Lowercase top for the prefix was correct, not sure why you changed it.

**Comment by jamesmintram** [comment]
> Lowercase top for the prefix was correct, not sure why you changed it.

Changed it where?

**Comment by linusg** [comment]
In the commit message - or maybe I'm misinterpreting it, at least the PR title has it right. Either way, please update.

**Comment by jamesmintram** [comment]
> In the commit message - or maybe I'm misinterpreting it, at least the PR title has it right. Either way, please update.

Done.

**Pull request merged**